backend/middlewares/auth-middleware.js

```diff
@@ -24,3 +25,28 @@ export const authenticationHandler = async (req, res, next) => {
     });
   }
 };
+
+export const adminHandler = async (req, res, next) => {
+  try {
+    const user = await User.findById(req.user._id);
+    if (!user)
```

Comment:
Can we optimize this bit by avoiding the db call? Ideally lets put all the required details in the token's payload (ig already done) and we extract it to verify the token in the `authenticationHandler`, lets pass this user payload extracted in this handler to our new `adminHandler` this way we can avoid the db call

Reply:
I guess we don't required the complete payload even, just we need to check the role, lets avoid the db call then, get the user object from the `authenticationHandler`

Reply:
The payload only contains name and id of the user.
I have changed the payload and removed the db call from the admin middleware

Reply:
Super @devsharmagit, Thank you